core/src/main/java/hudson/model/User.java

```diff
@@ -331,10 +347,24 @@ public synchronized void addProperty(@Nonnull UserProperty p) throws IOException
         } catch (DataAccessException e) {
             // seems like it's in the same boat as UserMayOrMayNotExistException
         }
+        
+        return null;
+    }
 
+    /**
+     * Creates an {@link Authentication} object that represents this user using the given userDetails
+     *
+     * @param userDetails Provided by {@link #getUserDetailsForImpersonation()}
+     * @see #getUserDetailsForImpersonation()
+     */
```

Comment:
It would be great to document the minimal access thing directly in Javadoc

Reply:
Second sentence on the userDetails param added as requested